src/Spice86.Core/Emulator/Devices/Timer/Timer.cs

```diff
@@ -35,7 +37,9 @@ public class Timer : DefaultIOPortHandler {
     // screen refresh
     private readonly Counter _vgaScreenRefreshCounter;
 
-    public Timer(State state, ILoggerService loggerService, DualPic dualPic, IVideoCard? vgaCard, CounterConfigurator counterConfigurator, bool failOnUnhandledPort) : base(state, failOnUnhandledPort, loggerService) {
+    private readonly IGui? _gui;
+
+    public Timer(IGui? gui, State state, ILoggerService loggerService, DualPic dualPic, IVideoCard? vgaCard, CounterConfigurator counterConfigurator, bool failOnUnhandledPort) : base(state, failOnUnhandledPort, loggerService) {
```

Comment:
This does not feel right, the Timer depending on the Gui, rather than the Gui depending on the Timer.
Is there a specific reason you set it up this way?

Reply:
Kevin said the same thing. This was to avoid any Law of Demeter violations in the ProgramExecutor. Most where resolved by the Visitor pattern, but not this one.

I didn't like the idea of using the Visitor pattern for everything. I'll think of another way (maybe a nullable ITimeMultiplier property that the Timer sets on creation).

The real fix would be to inject the timer into the GUI, but the the Timer depends on so many services that it becomes nightmarish to try and move all instanciations into Program.cs (where we create MainWindowViewModel).

Reply:
I now recall that I saw a WPF ChatGPT example where Microsoft's DI (IServicesCollection) was used with the WPF app host. I can do something similar with the AvaloniaUI app host and inject the Timer into the UI.

This could even potientally remove the Visitor pattern... Must think more about this. :)

Reply:
Fixed by using a property in IGui than the Machine sets once the Timer is created. The IGui doesn't inherit from INotifyPropertyChanged anymore either.